annotation/tests/test_jobs.py

```diff
@@ -343,4 +328,203 @@ def test_get_job_attributes_for_post_attribute_not_found():
+def test_create_job():
+    mock_session = MagicMock()
+    expected_result = Job(
+        job_id=1,
+        name="test1",
+        callback_url="http://www.test.com",
+        annotators=[None],
+        files=[None],
+        is_auto_distribution=False,
+        categories=[None],
+        deadline=None,
+        job_type=JobTypeEnumSchema.ExtractionJob,
+        tenant="test",
+    )
+    result = annotation.jobs.services.create_job(
+        mock_session,
+        JobInfoSchema(
+            job_id=1,
+            name="test1",
+            callback_url="http://www.test.com",
+            annotators=set(),
+            files={1},
+            validators=set(),
+            owners={uuid.UUID("82533770-a99e-4873-8b23-6bbda86b59ae")},
+            previous_jobs=[],
+            datasets={1, 2},
+            is_auto_distribution=False,
+            categories=set(),
+            deadline=None,
+            job_type=JobTypeEnumSchema.ExtractionJob,
+            tenant="test",
+        ),
+    )
+    assert result == expected_result
+
+
+def test_add_users(users: Tuple[User, ...]):
+    mock_session = MagicMock()
+    cur_users = [users[1], users[2]]
+    new_user_ids = {UUID(int=3), UUID(int=4)}
+    expected_result = [
+        users[1],
+        users[2],
+        users[3],
+    ]
+    mock_session.query().filter().all.return_value = [users[3]]
+    result = annotation.jobs.services.add_users(
+        mock_session, cur_users, new_user_ids
+    )
+    assert result == expected_result
+
+
+def test_update_job_categories(categories: Category):
+    mock_session = MagicMock()
+    patch_data = {"categories": [1, 2, 3]}
+    expected_result = ["18d3d189e73a4680bfa77ba3fe6ebee5"]
+    tenat = "test"
+    with patch(
+        "annotation.jobs.services.fetch_bunch_categories_db",
+        return_value=[categories],
+    ):
+        annotation.jobs.services.update_job_categories(
+            mock_session, patch_data, tenat
+        )
+        assert [cat.id for cat in patch_data["categories"]] == expected_result
+
+
+@pytest.mark.parametrize(
+    ("patch_data", "user_id"),
+    (
+        (
+            {
+                "extensive_coverage": 1,
+                "annotators": [
+                    User(user_id=UUID(int=1)),
+                    User(user_id=UUID(int=2)),
+                ],
+            },
+            (
+                3,
+                4,
+            ),
+        ),
+        (
+            {"extensive_coverage": 1, "annotators": []},
+            (
+                3,
+                4,
+            ),
+        ),
+        (
+            {
+                "extensive_coverage": 5,
+                "annotators": [
+                    User(user_id=UUID(int=1)),
+                    User(user_id=UUID(int=2)),
+                ],
+            },
+            (
+                3,
+                4,
+            ),
+        ),
+        (
+            {"extensive_coverage": 5, "annotators": []},
```

Comment:
this data is a perfect to use with parametrize permutations. 
https://www.youtube.com/watch?v=QIUrd327tOQ

Reply:
Watched the video and that is pretty useful, but in the end i split the test in two so did not use it in the end